packages/flutter/lib/src/gestures/hit_test.dart

```diff
@@ -54,30 +61,188 @@ class HitTestEntry {
 
 /// The result of performing a hit test.
 class HitTestResult {
-  /// Creates a hit test result.
-  ///
-  /// If the [path] argument is null, the [path] field will be initialized with
-  /// and empty list.
-  HitTestResult({ List<HitTestEntry> path })
-    : _path = path ?? <HitTestEntry>[];
-
   /// An unmodifiable list of [HitTestEntry] objects recorded during the hit test.
   ///
   /// The first entry in the path is the most specific, typically the one at
   /// the leaf of tree being hit tested. Event propagation starts with the most
   /// specific (i.e., first) entry and proceeds in order through the path.
-  Iterable<HitTestEntry> get path => _path;
-  final List<HitTestEntry> _path;
+  Iterable<HitTestEntry> get path => _path.keys;
+  final Map<HitTestEntry, Matrix4> _path = <HitTestEntry, Matrix4>{};
```

Comment:
would it make more sense to store the matrix on the entry itself?

Reply:
That would be another breaking change since `HitTestEntry` currently has a const constructor. If we would allow `HitTestResult` to set the appropriate transform on the `HitTestEntry` it can no longer be const. We could make it the responsibility of the HitTestEntry provider to set the proper transform when the entry is constructed - but that would be more error-prone.

I am leaning towards keeping it as-is, but if you prefer the non-const approach for `HitTestEntry` I am happy to change it.

Reply:
I'm not too worried about HitTestEntry not being const. Hit tests by definition can't be done at compile time. :-)